test/compiler.test.js

```diff
@@ -514,6 +514,37 @@ describe('compiler', function() {
       expect(instructions.files.boot).to.eql([appJs]);
     });
 
+    it('prefers coffeescript over json in `appRootDir/bootDir`', function() {
+      // add coffee-script to require.extensions
+      require('coffee-script/register');
+
+      appdir.createConfigFilesSync();
+      var coffee = appdir.writeFileSync('./custom-boot/init.coffee', '');
+      appdir.writeFileSync('./custom-boot/init.json', {});
+
+      var instructions = boot.compile({
+        appRootDir: appdir.PATH,
+        bootDirs: ['./custom-boot']
+      });
+      expect(instructions.files.boot).to.eql([coffee]);
+    });
+
+    it('prefers coffeescript over json in `node_modules/bootDir`', function() {
+      // add coffee-script to require.extensions
+      require('coffee-script/register');
+
+      appdir.createConfigFilesSync();
+      var coffee = appdir.writeFileSync('node_modules/custom-boot/init.coffee',
+        '');
+      appdir.writeFileSync('node_modules/custom-boot/init.json', '');
+
+      var instructions = boot.compile({
+        appRootDir: appdir.PATH,
+        bootDirs: ['node_modules/custom-boot']
```

Comment:
This is weird. If you want to test the module-based resolution of paths then you should use `bootDirs: ['custom-boot']` only.

I don't know if the option `bootDirs` supports paths pointing to modules. It's possible that such paths are not supported, in which case this test is not necessary.

Also please keep all `bootDirs` close test together, rather than having them intertwined with other tests (e.g. for `bootScripts`).